utils/buffer/examples/buffer_pool_exhaustion.rs

```diff
@@ -0,0 +1,65 @@
+// # Handling Buffer Pool Exhaustion and Heap Allocation
+//
+// This example demonstrates how a buffer pool is filled. The back slots of the buffer pool are
+// exhausted first, followed by the front of the buffer pool. Once both the back and front are
+// exhausted, data is allocated on the heap at a performance decrease.
+//
+// 1. Fills up the back slots of the buffer pool until they’re exhausted.
+// 2. Releases one slot to allow the buffer pool to switch to front mode.
+// 3. Fully fills the front slots of the buffer pool.
+// 4. Switches to alloc mode for direct heap allocation when both the buffer pool's back and front
+//    slots are at capacity.
+//
```

Comment:
```suggestion
// Below is a visual representation of how the buffer pool evolves evolves as the example progresses:
//
// --------  BACK MODE
// a-------  BACK MODE (add "a" via loop)
// aa------  BACK MODE (add "a" via loop)
// aaa-----  BACK MODE (add "a" via loop)
// aaaa----  BACK MODE (add "a" via loop)
// -aaa----  BACK MODE (pop front)
// -aaab---  BACK MODE (add "b")
// -aaabc--  BACK MODE (add "c" via loop)
// -aaabcc-  BACK MODE (add "c" via loop)
// -aaabccc  BACK MODE (add "c" via loop)
// caaabccc  BACK MODE (add "c" via loop, which gets added via front mode)
// caaabccc  ALLOC MODE (add "d", allocated in a new space in the heap)
```

Reply:
I think it is overkill since we have these cases in the `README.md`.

Reply:
the `README.md`has a similar table but it's not identical to this

the motivation was the fact that @GitGab19 was confused about the loops, so the goal of this one is to help the reader understand the flow of the example, especially what's happening around each loop